torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -3992,3 +3992,93 @@ def forward(_self, input, expected_type):  # noqa
 
             inp = TestNamedTupleInput_1(a, b)
             model(inp, type(inp))
+
+        @require_backend({"gloo", "nccl"})
+        @require_backends_available({"gloo", "nccl"})
+        @skip_if_lt_x_gpu(2)
+        @skip_if_rocm
+        def test_ddp_control_flow_same_across_ranks(self):
+            # Control flow that is the same across ranks.
+            class ToyModel(nn.Module):
+                def __init__(self):
+                    super(ToyModel, self).__init__()
+                    self.lin1 = nn.Linear(10, 10, bias=False)
+                    self.lin2 = nn.Linear(10, 10, bias=False)
+                    self.counter = 0
+
+                def forward(self, x):
```

Comment:
Ideally we should test input dependent control flows (conditionals on x).

Reply:
Updated to test based on `x`, basically we do a `torch.equal` using x and an expected tensor. Please let me know if there is a better way to test that.